### PR TITLE
redis: make Deployment roll out on configuration changes

### PR DESCRIPTION
**Pull request by stanislav-zaprudskiy**
The mechanics of how the annotation causes the deployment to roll out is explained here - https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments.

Without this PR any changes of Redis configuaration or password require additional intervention (e.g. manual `kubectl rollout restart deployment`) to make them picked up, which doesn't scale well on frequent changes or with numerous deployments.

### CODE REVIEW

File: common/redis/templates/deployment.yaml
Comment (SuperSandro2000) [inline common/redis/templates/deployment.yaml:37]: I've looked through our other things and sometimes we use Template.BasePath and sometimes not. I leave it up to @majewsky what we use.

I don't thinkwe need to be conditional here, worst case we hash the empty files.

```suggestion
        checksum/configmap: {{ include "redis/templates/configmap.yaml" . | sha256sum }}
        checksum/secrets: {{ include "redis/templates/secrets.yaml" . | sha256sum }}
```

Comment (majewsky) [reply common/redis/templates/deployment.yaml:37]: I believe the form with `.Template.BasePath` is necessary for library charts that are included as dependencies elsewhere, so the full path would be something like `my-app/charts/redis/templates/configmap.yaml`.

**Review by majewsky — CHANGES_REQUESTED** [review]
We can take the current form with `.Template.BasePath`, but please bump the version in Chart.yaml before merging.

**Review by majewsky — APPROVED** [review]

**Pull request merged**
